biz.aQute.bndlib/src/aQute/bnd/osgi/URLResource.java

```diff
@@ -32,9 +33,18 @@ private ByteBuffer getBuffer() throws Exception {
 			return buffer;
 		}
 		if (url.getProtocol().equals("file")) {
-			File file = new File(url.getPath());
+			File file = new File(url.toURI());
 			lastModified = file.lastModified();
 			return buffer = IO.read(file.toPath());
+		} else if (url.getProtocol().equals("jar")) {
+			String subUrl = url.toString();
+			int entryStart = subUrl.indexOf("!/");
+			String entry = subUrl.substring(entryStart + 2);
+			subUrl = subUrl.substring(4, entryStart);
+
+			try (Jar jar = Jar.fromResource("tmp", new URLResource(new URL(subUrl)))) {
```

Comment:
This seems like a bad performance idea. This reads the entire jar into memory. Perhaps this is why the build takes 5 min longer. I did a bunch of debugging on this and have a more complete and performant solution I will make a PR for.